docker/server/entrypoint.sh

```diff
@@ -61,7 +64,40 @@ do
     fi
 done
 
+# if clickhouse user is defined - create it (user "default" already exists out of box)
+if [ -n "$CLICKHOUSE_USER" ] && [ "$CLICKHOUSE_USER" != "default" ]; then
+  echo "Create user '$CLICKHOUSE_USER' instead default"
+  cat <<EOT >> /etc/clickhouse-server/users.d/default-user.xml
+  <yandex>
+    <!-- Docs: <https://clickhouse.tech/docs/en/operations/settings/settings_users/> -->
+    <users>
+      <!-- Remove default user -->
+      <default remove="remove">
+      </default>
+
+      <${CLICKHOUSE_USER}>
+        <profile>default</profile>
+        <networks>
+          <ip>::/0</ip>
+        </networks>
+        <password>${CLICKHOUSE_PASSWORD}</password>
+        <quota>default</quota>
+      </${CLICKHOUSE_USER}>
+    </users>
+  </yandex>
+EOT
+fi
 
+# define password argument for clickhouse client
+if [ -n "$CLICKHOUSE_PASSWORD" ]; then
+    printf -v WITH_PASSWORD '%s %q' "--password" "$CLICKHOUSE_PASSWORD"
+fi
+
+# create default database, if defined
+if [ -n "$CLICKHOUSE_DB" ]; then
+  echo "Create database '$CLICKHOUSE_DB'"
+  clickhouse-client --query -u "$CLICKHOUSE_USER" $WITH_PASSWORD "CREATE DATABASE IF NOT EXISTS $CLICKHOUSE_DB";
```

Comment:
You should also move `clickhouseclient` variable higher and you it here

Reply:
Database creation must be called **after** database "ping-ing". So, lines with `clickhouseclient` and password (with this line) moved back